authenticate-oauth/Web/Authenticate/OAuth.hs

```diff
@@ -355,12 +375,21 @@ injectOAuthToCred oa cred =
             ] cred
 
 
+-- Note that prefix is used for realm in addAuthHeader, so should fix that up,
+-- but may need an API break for that.
+addAuthBody :: BS.ByteString -> Credential -> Request -> Request
+addAuthBody _ (Credential cred) req = urlEncodedBody (filterCreds cred) req
```

Comment:
*Now* I understand what you're saying, I didn't realize you were using this as the parameter to `getAccessToken''` (you should know that I only recently took over maintainership of this library, and have actually never used it myself). OK, that makes sense. The way you could do it is `const addAuthBody`... but the way you've written it now is probably better. I'd recommend:

1. Making the first parameter a simple `a` instead of hard-coding to `BS.ByteString`.
2. Adding a Haddock comment explaining that the argument is not used, and that it's present simply to give the same type as `addAuthHeader`.